dev/src/External/CashierDesk.java

```diff
@@ -66,21 +64,28 @@ public void turnOn() throws Exception {
                     break;
                 }
             }
-
         }
     }
 
-    private boolean chooseStore() {
-        this.out.println("There are: " + this.controllerRef.getStores().size() + " stores");
-        this.out.print("Please enter store id: ");
-        int storeIndex = Integer.parseInt(this.in.nextLine());
-        if (storeIndex < this.controllerRef.getStores().size())
-            chosenStore = this.controllerRef.getStores().get(storeIndex);
-        else {
-            this.out.println("Invalid store id\nReturning to main menu...\n");
+
+    private boolean displayStores() {
+        StoresScreen storesScreen = new StoresScreen(this.out, this.in, controllerRef);
+        int userInput = storesScreen.handleMsg();
+
+        if (userInput >= controllerRef.getStores().size() || userInput < 0) {
+            this.out.println(SuppliersConstants.INVALID_INPUT);
```

Comment:
ההודעה תקפוץ גם אם אין חנויות בכלל
![image](https://github.com/user-attachments/assets/f3366496-65de-4902-afe9-fa042aa2af5b)

Reply:
fixed in #50